control-plane/connect-inject/common/common.go

```diff
@@ -12,6 +12,40 @@ import (
 	corev1 "k8s.io/api/core/v1"
 )
 
+// DetermineAndValidatePort behaves as follows:
+// If the annotation exists, validate the port and return it.
+// If the annotation does not exist, return the default port.
+// If the privileged flag is true, it will allow the port to be in the
+// privileged port range of 1-1023. Otherwise, it will only allow ports in the
+// unprivileged range of 1024-65535.
+func DetermineAndValidatePort(pod corev1.Pod, annotation string, defaultPort string, privileged bool) (string, error) {
```

Comment:
Moved this into `common` as it's now used by both `metrics` and `lifecycle` packages.